modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java

```diff
@@ -0,0 +1,379 @@
+			editableValuesJSON, "$GROUP_ID",
+			String.valueOf(_group.getGroupId()));
+
+		replacedValues = StringUtil.replace(
+			replacedValues, "$LAYOUT_ID",
+			String.valueOf(layout4.getLayoutId()));
+		replacedValues = StringUtil.replace(
+			replacedValues, "$LAYOUT_UUID", String.valueOf(layout4.getUuid()));
+		replacedValues = StringUtil.replace(
+			replacedValues, "$TITLE", layout4.getName("en_US"));
+
+		fragmentEntryLink.setEditableValues(replacedValues);
+
+		fragmentEntryLink.setSegmentsExperienceId(
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				masterPageTemplateLayout.getPlid()));
+
+		fragmentEntryLink.setGroupId(_group.getGroupId());
+
+		_fragmentEntryLinkLocalService.updateFragmentEntryLink(
+			fragmentEntryLink);
+
+		long[] layoutIds = {
+			layout1.getLayoutId(), layout2.getLayoutId(), layout3.getLayoutId(),
+			layout4.getLayoutId()
+		};
+
+		exportImportLayouts(layoutIds, _getImportParameterMap());
+
+		Layout importedLayout1 =
+			LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
+				layout1.getUuid(), _importedGroup.getGroupId(), false);
+
+		Layout importedLayout2 =
+			LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
+				layout2.getUuid(), _importedGroup.getGroupId(), false);
+
+		Layout importedLayout3 =
+			LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
+				layout3.getUuid(), _importedGroup.getGroupId(), false);
+
+		Layout importedLayout4 =
+			LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
+				layout4.getUuid(), _importedGroup.getGroupId(), false);
+
+		Layout importedMasterPageTemplateLayout =
+			LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
+				masterPageTemplateLayout.getUuid(), _importedGroup.getGroupId(),
+				true);
+
+		Layout importedDraftLayout1 = importedLayout1.fetchDraftLayout();
+		Layout importedDraftLayout2 = importedLayout2.fetchDraftLayout();
+		Layout importedDraftLayout3 = importedLayout3.fetchDraftLayout();
+		Layout importedDraftLayout4 = importedLayout4.fetchDraftLayout();
+		Layout importedDraftLayoutOfMasterPageTemplate =
+			importedMasterPageTemplateLayout.fetchDraftLayout();
+
+		Assert.assertTrue(importedDraftLayout1.isDraftLayout());
+		Assert.assertEquals(
+			importedLayout1.getName(), importedDraftLayout1.getName());
+
+		Assert.assertTrue(importedDraftLayout2.isDraftLayout());
+		Assert.assertEquals(
+			importedLayout2.getName(), importedDraftLayout2.getName());
+
+		Assert.assertTrue(importedDraftLayout3.isDraftLayout());
+		Assert.assertEquals(
+			importedLayout3.getName(), importedDraftLayout3.getName());
+
+		Assert.assertTrue(importedDraftLayout4.isDraftLayout());
+		Assert.assertEquals(
+			importedLayout4.getName(), importedDraftLayout4.getName());
+
+		Assert.assertTrue(
+			importedDraftLayoutOfMasterPageTemplate.isDraftLayout());
+		Assert.assertEquals(
+			importedMasterPageTemplateLayout.getName(),
+			importedDraftLayoutOfMasterPageTemplate.getName());
+	}
+
+	protected void exportImportLayouts(
```

Comment:
All these should be private methods. Protected methods should only be present in BaseTest classes that are meant to be extended.